### PR TITLE
Feature: add revenue compatibility for v3 forms and donation model

### PR DESCRIPTION
**Pull request by jonwaldstein**
Resolves [GIVE-178]

## Description

This is necessary for updating the revenue amount in v3 forms that use models like give-funds.  Most notably for use with fee-recovery as the amount gets updated after initial donation creation. In v2 forms this was all processed during the initial instance of `give_insert_payment` - where in v3 it's more of an async request and we need to listen for udpates.

Basically, this will ensure the revenue table gets updated when a donation model is updated.

## Affects

- Updated an existing action to accept a donation model instead of donationId which is now used across both `give_updated_edited_donation` and `givewp_donation_updated` for the purpose of updating revenue amounts.

## Visuals

Screenshot 2023-12-14 at 1 01 22 PM

Screenshot 2023-12-14 at 1 03 07 PM

## Testing Instructions

- Make sure v2 forms still work properly with fee recovery and funds
- v3 and funds compatibility is still in progress so would either need to test pending PR or programatically

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

[GIVE-178]: https://stellarwp.atlassian.net/browse/GIVE-178?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by JasonTheAdams** [comment]
@jonwaldstein I can't remember how our model compatibility works. By switching to the model "updated" hook, will this still work sufficiently with v2 forms? Or is there a reason that's not necessary?

**Comment by jonwaldstein** [comment]
@JasonTheAdams that's a great question!

We are using the models in _both_ v2 and v3 forms - so the model hooks _will_ fire for both forms.  In fact, we have done our best to make sure the processing is as close as possible.  However, there is still a difference in how some data is sent to the server and how we process custom fields in v2 forms that we need to account for.  

For example, in v2 forms we send the sum amount of fee recovered to the server, where as in v3 forms, the two amounts are delivered separately and updated on the server via hooks.

So to answer your question, yes the "updated" hook will work for v2 forms - however we are not exactly _switching_ to that hook - it's being added in addition to `give_insert_payment` and `give_updated_edited_donation` (admin).  It's possible we can replace `give_insert_payment` with `givewp_donation_created` but i'm not ready to test that theory at the moment since we're still using the former throughout addons.

**Comment by jonwaldstein** [comment]
@glaubersilva can you please review this one

**Review by glaubersilva — APPROVED** [review]
@jonwaldstein Nice work! Everything worked as expected for both v2 and v3 forms. I just left a  couple of comments about a few minor changes related to the missing unreleased tag. After that, it's ready for QA. Thanks!

File: tests/Unit/Revenue/LegacyListeners/UpdateRevenueWhenDonationAmountUpdatedTest.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+use Give\Donations\Models\Donation;
+use Give\Donations\ValueObjects\DonationStatus;
+use Give\Framework\Database\DB;
+use Give\Framework\Support\ValueObjects\Money;
+use Give\Revenue\LegacyListeners\UpdateRevenueWhenDonationAmountUpdated;
+use Give\Tests\TestCase;
+use Give\Tests\TestTraits\RefreshDatabase;
+
+class UpdateRevenueWhenDonationAmountUpdatedTest extends TestCase
```
[the remark this answers is not in the document]
Comment (jonwaldstein) [reply tests/Unit/Revenue/LegacyListeners/UpdateRevenueWhenDonationAmountUpdatedTest.php:16]: Resolved 35981fa8

**Review by rickalday — APPROVED** [review]
Passed manual QA tests.

**Pull request merged**